pyteal/ast/router.py

```diff
@@ -459,34 +560,92 @@ def wrap_handler(
                     abi.MethodReturn(output_temp),
                     Approve(),
                 )
+        else:
+            subroutine_ify = ASTBuilder.__de_abify_subroutine(handler)
+            return Seq(
+                subroutine_ify(),
+                Approve(),
+            )
+
+    @staticmethod
+    def __de_abify_subroutine(
+        handler: ABIReturnSubroutine | SubroutineFnWrapper | Expr,
+    ) -> SubroutineFnWrapper:
+        handler = ASTBuilder.__handler_validity_check(handler)
+        (
+            arg_vals,
+            app_arg_vals,
+            txn_arg_vals,
+        ) = ASTBuilder.__subroutine_argument_instance_generate(handler)
+
+        (
+            decode_instructions,
+            arg_vals,
+            proto,
+        ) = ASTBuilder.__decode_constructions_and_args(
+            arg_vals,
+            app_arg_vals,
+            txn_arg_vals,
+            handler,
+            use_frame_pt=True,
+        )
+
+        subroutine_caster = Subroutine(TealType.none, f"{handler.name()}_caster")
+        return_expr: Expr
+
+        if not proto or not proto.mem_layout:
```

Comment:
Is there a case that `ASTBuilder.__decode_constructions_and_args()` may return an invalid such `proto` ? If so, it should probably in charge of this assertion (for failing fast's sake) especially since it has access to the `use_frame_ptr` parameter.

Reply:
I somehow doubt why this check is where, generally `ASTBuilder.__decode_constructions_and_args()` should not return an invalid `proto`, for in our case, we want `proto 0 0` with some memory layouts.

I will simply annihilate this unnecessary check.